external/implementations/samlconf-openam-impl/src/main/java/org/codice/compliance/saml/plugin/openam/OpenAMIdpSSOResponderProvider.java

```diff
@@ -0,0 +1,168 @@
+/*
+Copyright (c) 2018 Codice Foundation
+
+Released under the GNU Lesser General Public License; see
+http://www.gnu.org/licenses/lgpl.html
+*/
+package org.codice.compliance.saml.plugin.openam;
+
+import com.google.common.base.Splitter;
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import io.restassured.RestAssured;
+import io.restassured.response.Response;
+import java.io.UnsupportedEncodingException;
+import java.net.URLDecoder;
+import java.nio.charset.StandardCharsets;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Map;
+import org.codice.compliance.saml.plugin.IdpSSOResponder;
+import org.codice.compliance.utils.ConstantsKt;
+import org.codice.compliance.utils.GlobalSession;
+
+public class OpenAMIdpSSOResponderProvider implements IdpSSOResponder {
+
+  private static String TOKEN_ID = "tokenId";
+  private static String CALLBACKS = "callbacks";
+  private OpenAMDataParser openAMData = new OpenAMDataParser();
+
+  @Override
+  public Response getResponseForRedirectRequest(Response originalResponse) {
+    return openAMLogin(originalResponse);
+  }
+
+  @Override
+  public Response getResponseForPostRequest(Response originalResponse) {
+    return openAMLogin(originalResponse);
+  }
+
+  private Response openAMLogin(Response res) {
+
+    //  The Authentication Realm to connect to
+    String authURL =
+        openAMData.getBaseURL()
+            + "/json/realms/root"
+            + openAMData.getRealmsUrl()
+            + openAMData.getOpenAMRealm()
+            + "/authenticate";
+
+    Map<String, String> queryParams = getQueryParams(res.getHeader(ConstantsKt.LOCATION));
+
+    // -- Get the initial json form to fill in --
+    JsonParser parser = new JsonParser();
+    Response payloadRes = RestAssured.given().queryParams(queryParams).when().post(authURL);
+
+    JsonObject payload = parser.parse(payloadRes.jsonPath().prettify()).getAsJsonObject();
```

Comment:
❓ Is there a reason to call `prettify` here? Are we logging the pretty-print response?

Reply:
The reason was to act as a toString for the gson json parser. I liked the output of that more than the Hashmap returned from `payloadRes.jsonPath.getJsonObject("")`

Reply:
Although `toJsonString` would be a better method to call than `prettify`